shiny/_main.py

```diff
@@ -626,3 +629,47 @@ class ReloadArgs(TypedDict):
     reload_includes: NotRequired[list[str]]
     reload_excludes: NotRequired[list[str]]
     reload_dirs: NotRequired[list[str]]
+
+
+# Check that the version of rsconnect supports Shiny Express; can be removed in the
+# future once this version of rsconnect is widely used. (Added 2024-03)
+def _verify_rsconnect_version() -> None:
+    PACKAGE_NAME = "rsconnect-python"
+    MIN_VERSION = "1.22.0"
+
+    from importlib.metadata import PackageNotFoundError, version
+
+    def _safe_int(x: str) -> int:
+        try:
+            return int(x)
+        except ValueError:
+            return 0
+
+    def _compare_package_versions(version1: str, version2: str) -> Literal[-1, 0, 1]:
+        parts1 = [_safe_int(x) for x in version1.split(".")]
+        parts2 = [_safe_int(x) for x in version2.split(".")]
+
+        max_length = max(len(parts1), len(parts2))
+        parts1 += [0] * (max_length - len(parts1))
+        parts2 += [0] * (max_length - len(parts2))
+
+        for part1, part2 in zip(parts1, parts2):
+            if part1 > part2:
+                return 1
+            elif part1 < part2:
+                return -1
+
+        return 0
+
+    try:
+        package_version = version(PACKAGE_NAME)
+        print(package_version)
```

Comment:
Errant print?

Reply:
Oh oops, I had deleted it on my local copy of the file, but forgot to commit. 🤦